src/mscorlib/src/System/Number.cs

```diff
@@ -685,6 +685,14 @@ internal unsafe struct NumberBuffer {
 
         [System.Security.SecuritySafeCritical]  // auto-generated
         internal unsafe static Int32 ParseInt32(String s, NumberStyles style, NumberFormatInfo info) {
+            if (style == NumberStyles.Integer && info == NumberFormatInfo.InvariantInfo) {
```

Comment:
I see. I didn't know that `NumberFormatInfo` implemented `IFormatProvider` -- I never even knew you could pass that directly!
I'm not very knowledgeable about cultures and globalization, so I don't really know if the check can even be relaxed, or how to relax it in practice. A hint from someone who's more familiar in these areas would be nice. I definitely think it should be relaxed before this change is made.